### PR TITLE
improve documentation of OffsetBox

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/offsetbox.py [hunk 1]
```diff
@@ -164,6 +166,29 @@ def axes(self, ax):
                 c.axes = ax
 
     def contains(self, mouseevent):
+        """
+        Delegate the mouse event contains-check to the children.
+
+        As a container, the `.OffsetBox` itself does not respond to
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: does not respond itself

File: lib/matplotlib/offsetbox.py [hunk 2]
```diff
@@ -188,58 +215,78 @@ def offset(width, height, xdescent, ydescent, renderer) \
 
     def get_offset(self, width, height, xdescent, ydescent, renderer):
         """
-        Get the offset
+        Return the offset as a tuple (x, y).
+
+        The extent parameters have to be provided in case the offset is
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: in case -> to handle the case where

File: lib/matplotlib/offsetbox.py [hunk 3]
```diff
@@ -458,20 +518,33 @@ def get_extent_offsets(self, renderer):
...
 
-        .. note::
-          *pad* need to given in points and will be
-          scale with the renderer dpi, while *width* and *height*
-          need to be in pixels.
-        """
+        pad : float
+            The padding in points. This will be scaled with the renderer dpi.
+            In contrast *width* and *hight* are in *pixel* and thus not scaled.
+
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: skip the extra newlines?

File: lib/matplotlib/offsetbox.py [hunk 4]
```diff
@@ -319,23 +372,29 @@ def __init__(self, pad=None, sep=None, width=None, height=None,
...
-            Boundary pad.
+            The boundary padding in points.
 
         sep : float, optional
-            Spacing between items.
+            The spacing between items in points.
 
-        width : float, optional
+        width, height : float, optional
+           Width and height of the container box in pixels, calculated if
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: this paragraph's indent is off by 1

File: lib/matplotlib/offsetbox.py [hunk 5]
```diff
@@ -271,23 +318,29 @@ def __init__(self, pad=None, sep=None, width=None, height=None,
...
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: this paragraph's indent is off by 1

File: lib/matplotlib/offsetbox.py [hunk 6]
```diff
@@ -395,22 +451,29 @@ def __init__(self, pad=None, sep=None, width=None, height=None,
...
         pad : float, optional
-            Boundary pad.
+            The boundary padding in points.
 
         sep : float, optional
-            Spacing between items.
+            The spacing between items in points.
+
+        width, height : float, optional
+           Width and height of the container box in pixels, calculated if
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py]: indent still off by 1 :)

File: lib/matplotlib/axes/_base.py [hunk 7]
```diff
@@ -2428,6 +2428,7 @@ def autoscale_view(self, tight=None, scalex=True, scaley=True):
         case, use :meth:`matplotlib.axes.Axes.relim` prior to calling
         autoscale_view.
         """
+        print('asv', scalex, scaley)
```
Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: @timhoffm I missed these prints :(
Can you revert them?  Thanks.

Comment (timhoffm) [reply lib/matplotlib/axes/_base.py]: Woops, will do tonight.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.1.x

**Pull request merged**